src/events.js

```diff
@@ -1,22 +1,50 @@
 import {empty} from 'most'
 import fromEvent from './fromEvent'
+import {makeIsStrictlyInRootScope} from './select'
 
-const makeEventsSelector =
-  element$ =>
-    (type, useCapture = false) => {
-      if (typeof type !== `string`) {
-        throw new Error(`DOM drivers events() expects argument to be a ` +
-          `string representing the event type to listen for.`)
-      }
-      return element$
-        .map(
-          elements =>
-            elements ?
-              fromEvent(type, elements, useCapture) :
-              empty()
-        )
-        .switch()
-        .multicast()
+let matchesSelector
+try {
+  matchesSelector = require(`matches-selector`)
+} catch (err) {
+  matchesSelector = () => {}
+}
+
+function makeEventsSelector(element$, selector) {
+  return function eventsSelector(type, useCapture = false) {
+    if (typeof type !== `string`) {
+      throw new Error(`DOM drivers events() expects argument to be a ` +
+        `string representing the event type to listen for.`)
     }
+    return element$
+      .map(element => {
+        return {element, selector}
+      })
+      .skipRepeatsWith((prev, curr) => {
+        return prev.selector.join(``) === curr.selector.join(``)
+      })
+      .map(({element}) => element)
+      .map(elements => {
```

Comment:
@TylorS since Motorcycle is about performance, I think you can join these two maps here, it's unnecessary overhead.